app/src/main/java/ai/brokk/analyzer/PythonAnalyzer.java

```diff
@@ -563,21 +602,21 @@ protected Set<CodeUnit> resolveImports(ProjectFile file, List<String> importStat
                                     decls.stream()
                                             .filter(cu -> cu.identifier().equals(text) && cu.isClass())
                                             .findFirst()
-                                            .ifPresent(resolved::add);
+                                            .ifPresent(cu -> resolvedByName.put(cu.identifier(), cu));
                                 } catch (Exception e) {
                                     log.warn(
                                             "Could not resolve import '{}' from module {}: {}",
                                             text,
                                             currentModule,
                                             e.getMessage());
                                 }
-                            } else if (currentModule == null) {
-                                // For "import X" style
+                            } else if (wildcardModule == null) {
+                                // For "import X" style (no module context at all)
                                 var definitions = getDefinitions(text);
                                 definitions.stream()
                                         .filter(CodeUnit::isClass)
```

Comment:
This brings up an interesting question - so Python would import a whole "module" which may include fields, classes, functions, etc. 

I recall you mentioned doing exportable functions/fields in a follow-up, so I'm happy to ignore that for now but it may be good to leave an explicit comment so that it's easy to find places like this that explicitly filter "future work" out.